src/test/database/fakeListRemote.spec.ts

```diff
@@ -0,0 +1,108 @@
+import * as pathLib from "path";
+import { expect } from "chai";
+
+import { ListRemote } from "../../database/listRemote";
+
+/**
+ * `FakeListRemote` is a test fixture for verifying logic lives in the
+ * `DatabaseRemove` class. It is essentially a mock for the Realtime Database
+ * that accepts a JSON tree to serve upon construction.
+ */
+export class FakeListRemote implements ListRemote {
+  data: any;
+  delay: number;
+
+  /**
+   * @param data the fake database structure. Each leaf is an integer
+   *   representing the subtree's size.
+   * @param largeThreshold the threshold to determine if a delete exceeds the
+   *   writeSizeLimit. If the sum of all leaves to delete is larger than
+   *   largeThreshold the delete will return false.
+   */
+  constructor(data: any) {
+    this.data = data;
+    this.delay = 0;
+  }
+
+  async listPath(
+    path: string,
+    numChildren: number,
+    startAfter?: string,
+    timeout?: number
+  ): Promise<string[]> {
+    const startTime = Date.now();
+    await new Promise((resolve, _) => setTimeout(resolve, this.delay));
```

Comment:
Not convinced this is the right way to fake timeout.
Unit test should be fast. If anyone use `this.delay`, it will make CI slower.

Potential alternative:
1. `timeout = 0` => always timeout
2. Otherwise, assume do not timeout.